pandas/io/sql.py

```diff
@@ -2117,7 +2165,8 @@ def to_sql(
         chunksize=None,
         dtype: DtypeArg | None = None,
         method=None,
-        **kwargs,
```

Comment:
Could you tackle this change in a separate PR? It looks like this also needs to be fixed in `pandas/core/generic.py` where users first interact with `to_sql`

Reply:
This change sprung out of making `SQLDatabase` method arguments match `SQLiteDatabase`, which helped me resolve mypy errors.  The `**kwargs` is replaced with the `engine` keyword argument + `**engine_kwargs`, which is what `SQLDatabase.to_sql` expects.  But I could split this out if you prefer.

Reply:
@mroeschke This change was made to the `SQLDatabase` class as a first step towards implementing #36893.  Currently, `to_sql` is only supported via `sqlalchemy`, but other engines might be added in the future.  So I think it makes sense to not change the signature of `to_sql` in `pandas/core/generic.py`, since end users cannot currently do anything with this functionality.  Let me know what you think.

Reply:
I split out this change in #49587 .